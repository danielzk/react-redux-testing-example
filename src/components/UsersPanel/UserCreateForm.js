import React from 'react'

export default class UserCreateForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      firstName: '',
      lastName: '',
    }
  }

  onUsernameChange = (event) => {
    this.setState({username: event.target.value})
  }

  onFirstNameChange = (event) => {
    this.setState({firstName: event.target.value})
  }

  onLastNameChange = (event) => {
    this.setState({lastName: event.target.value})
  }

  onSubmit = (event) => {
    this.props.actions.createUser(
      this.state.username, this.state.firstName, this.state.lastName)
    event.preventDefault()
  }

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <input
          id="username"
          type="text"
          value={this.state.username}
          onChange={this.onUsernameChange}
          placeholder="Username"
        />

        <input
          id="first-name"
          type="text"
          value={this.state.firstName}
          onChange={this.onFirstNameChange}
          placeholder="First name"
        />

        <input
          id="last-name"
          type="text"
          value={this.state.lastName}
          onChange={this.onLastNameChange}
          placeholder="Last name"
        />

        <button type="submit">Create</button>
      </form>
    )
  }
}
