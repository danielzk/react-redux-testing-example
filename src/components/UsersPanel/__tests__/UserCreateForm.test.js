import React from 'react'
import {shallow} from 'enzyme'

import UserCreateForm from '../UserCreateForm'

describe('UserCreateForm component', () => {
  it('should render field values', () => {
    const wrapper = shallow(<UserCreateForm/>)
    wrapper.setState({
      username: 'username',
      firstName: 'Daniel',
      lastName: 'Ramos',
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('should update username state when username input is changed', () => {
    let wrapper = shallow(<UserCreateForm/>)
    const event = {target: {value: 'usern'}}
    wrapper.find('#username').simulate('change', event)
    wrapper = wrapper.update()
    expect(wrapper.state('username')).toBe('usern')
  })

  it('should update firstName state when firstName input is changed', () => {
    let wrapper = shallow(<UserCreateForm/>)
    const event = {target: {value: 'Daniel'}}
    wrapper.find('#first-name').simulate('change', event)
    wrapper = wrapper.update()
    expect(wrapper.state('firstName')).toBe('Daniel')
  })

  it('should update lastName state when firstName input is changed', () => {
    let wrapper = shallow(<UserCreateForm/>)
    const event = {target: {value: 'Ramos'}}
    wrapper.find('#last-name').simulate('change', event)
    wrapper = wrapper.update()
    expect(wrapper.state('lastName')).toBe('Ramos')
  })

  it('should call createUser when form is submitted', () => {
    const props = {
      actions: {
        createUser: jest.fn(),
      }
    }
    const wrapper = shallow(<UserCreateForm {...props}/>)
    wrapper.setState({
      username: 'username',
      firstName: 'Daniel',
      lastName: 'Ramos',
    })

    const event = {preventDefault: jest.fn()}
    wrapper.find('form').simulate('submit', event)
    expect(props.actions.createUser).toHaveBeenCalledTimes(1)
    expect(props.actions.createUser).toBeCalledWith('username', 'Daniel', 'Ramos')
    expect(event.preventDefault).toHaveBeenCalledTimes(1)
  })
})
