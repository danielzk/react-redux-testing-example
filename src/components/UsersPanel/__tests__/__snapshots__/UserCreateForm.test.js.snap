// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCreateForm component should render field values 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
    value="username"
  />
  <input
    id="first-name"
    onChange={[Function]}
    placeholder="First name"
    type="text"
    value="Daniel"
  />
  <input
    id="last-name"
    onChange={[Function]}
    placeholder="Last name"
    type="text"
    value="Ramos"
  />
  <button
    type="submit"
  >
    Create
  </button>
</form>
`;
